frontend/apps/reader/modules/readersearch.lua

```diff
@@ -173,29 +304,46 @@ function ReaderSearch:onShowSearchDialog(text, direction)
         -- Keep the LTR order of |< and >|:
         from_start_text, from_end_text = BD.ltr(from_end_text), BD.ltr(from_start_text)
     end
+    self.wait_button = ButtonDialog:new{
+        buttons = {{{ text = "⌛" }}},
+    }
```

Comment:
Just noticed that if you have moved the search arrow button bar, this still gets displayed in the center of screen, and not over the arrow bar.

Reply:
This should solved it:
```diff
--- a/frontend/apps/reader/modules/readersearch.lua
+++ b/frontend/apps/reader/modules/readersearch.lua
@@ -311,6 +311,7 @@ function ReaderSearch:onShowSearchDialog(text, direction, regex, case_insensitiv
         if regex and isSlowRegex(pattern) then
             return function()
                 self.wait_button.alpha = 0.75
+                self.wait_button.movable:setMovedOffset(self.search_dialog.movable:getMovedOffset())
                 UIManager:show(self.wait_button)
                 UIManager:tickAfterNext(function()
                     do_search(func, pattern, param, regex, case_insensitive)()
@@ -355,6 +356,7 @@ function ReaderSearch:onShowSearchDialog(text, direction, regex, case_insensitiv
     }
     if regex and isSlowRegex(text) then
         self.wait_button.alpha = nil
+        self.wait_button.movable:setMovedOffset(self.search_dialog.movable:getMovedOffset())
         UIManager:show(self.wait_button)
         UIManager:tickAfterNext(function()
             do_search(self.searchFromCurrent, text, direction, regex, case_insensitive)()
```

Reply:
Ohh yes. I will look into it this evening.

Reply:
(The fix is above :) feel free to drop it in one of your related PRs)

Reply:
I think it is better to wait with this until #7956 is merged.